### PR TITLE
feat(Table): add sticky modifiers to Td

### LINKED ISSUE #8322: Sticky table column and disabled select
**Is this a new component or an extension of an existing one?**
Existing composable table

**Describe the feature**
1. In order to move Cost Management from the legacy table to the composable table, we need the first column to be sticky. In addition, we need the ability to disable the row's select component.

Unfortunately, the sticky column examples use a `Th` (to add `isStickyColumn`) instead of `Td` for the first column.
However, we cannot do that because `ThSelectType` does not support the same properties as `TdSelectType`. For example, `ThSelectType` is missing `disabled`, `rowIndex`, etc.

For our scenario,we just need the additional `disabled` support since we're using a bulk select.

**Cost management UI**
Screen Shot 2022-11-02 at 1 44 38 PM

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #8322

Allows `Td` to be a sticky column in the same manner as `Th` (testable by swapping the Th for Td in the table body of the sticky column example).

Also adds the disabled and additional properties to `ThSelectType` to match `TdSelectType`, which would let a user keep using `Th` for a select sticky column.

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]
Looks good! Regarding Dallas' question, it could be worth updating the description example

**Comment by kmcfaul** [comment]
Updated the description

File: packages/react-table/src/components/Table/base/types.tsx [hunk 1]
```diff
@@ -171,6 +171,10 @@ export interface ThSelectType {
   isSelected: boolean;
   /** Flag indicating the select checkbox in the th is disabled */
   isHeaderSelectDisabled?: boolean;
+  /** Whether to disable the selection */
+  disable?: boolean;
```
Comment (tlabaj) [inline packages/react-table/src/components/Table/base/types.tsx:175]: maybe isDisabled?

Comment (kmcfaul) [reply packages/react-table/src/components/Table/base/types.tsx]: After slack conversation - updating to `isDisabled` and opened #8419 to update `TdSelectType` to match as a breaking change

**Review by tlabaj — APPROVED** [review]

**Review by gitdallas — APPROVED** [review]

**Review by mattnolting — APPROVED** [review]
LGTM👍

**Pull request merged**